src/main/java/qna/domain/Question.java

```diff
@@ -68,25 +48,43 @@ public Question writeBy(User loginUser) {
 
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
-        answers.add(answer);
+        answers.addAnswer(answer);
     }
 
-    public boolean isOwner(User loginUser) {
-        return writer.equals(loginUser);
+
+    public DeleteHistories delete(User loginUser) throws CannotDeleteException {
+        if (loginUser != writer) {
+            throw new CannotDeleteException("질문자와 답변자가 달라서 삭제할수 없습니다.");
+        }
+        if (isAnswersOwner()) {
+            throw new CannotDeleteException("질문의 작성자가 아닌 유저가 답변을 달았을경우 삭제가 불가능합니다.");
+        }
+
+        DeleteHistories deleteHistories = DeleteHistories.of();
+        deleteHistories.save( deleteQuestion());
+        deleteHistories.saveAll(deleteAnswers(loginUser));
+
+        return deleteHistories;
```

Comment:
```suggestion
        return Stream.concat(
                    Stream.of(deleteQuestion()),
                    deleteAnswers(loginUser).stream())
                .collect(collectingAndThen(toList(), DeleteHistories::of));
```
이렇게 표현할 수 있을 것 같아요!
그리고 DeleteHistories의 불필요한 메소드는 지울 수 있지 않을까요?